lib/orchid/routing.rb

```diff
@@ -1,41 +1,30 @@
-# Load routes into Orchid::Routing::ROUTES constant before main app's routing
-require_relative "../../config/routes"
-
 module Orchid
-  class Routing
-
-    # Indicate draw method below available in routes.rb when run by Rails
-    @@ready_to_draw = true
+  module Routing
+    module_function
 
-    # Only draw Orchid routes once: before, between, or after main app's routes
-    @@routes_drawn = false
+    def draw(reserved_names: [])
+      # Retrieve list of main app's route names
+      drawn_routes = defined?(Rails.application.routes) ?
+        Rails.application.routes.routes.map { |r| r.name } : []
 
-    def self.draw(reserved_names: [])
-      if !@@routes_drawn
+      # If home path drawn, assume Orchid's routes have already been drawn
+      if !drawn_routes.include?("home")
         Rails.application.routes.draw do
-          # Retrieve list of main app's route names
-          app_route_names = Rails.application.routes.routes.map { |r| r.name }
-
           # Add names reserved by main app for more general routes, e.g. '/:id'
-          app_route_names += reserved_names
+          drawn_routes += reserved_names
 
           # Handle apps with relative url root
           url_prefix = Rails.application.config.relative_url_root.present? ?
```

Comment:
Is url_prefix still being used?

Reply:
It's not. Thanks for catching. I'll remove this.

Reply:
That project's override requirements are indeed met. It's deployed successfully with Passenger on the dev host.